fips.md

```diff
@@ -0,0 +1,80 @@
+
+> :warning: When using FIPS mode using different password for specific keystore elements is not supported and `key_pass` configuration field is ignored.
+
+2. Configure readonlyrest.yml to use new keystore and truststore. You will also need to add new configuration parameter `fips_mode`. Here's an example:
+``` 
+readonlyrest:
+  fips_mode: SSL_ONLY
+  ssl:
+    enable: true
+    keystore_file: "keystore.bcfks"
+    keystore_pass: readonlyrest
+    truststore_file: "truststore.bcfks"
+    truststore_pass: readonlyrest
+
+  ssl_internode:
+    enable: true
+    keystore_file: "keystore.bcfks"
+    keystore_pass: readonlyrest
+    key_pass: readonlyrest
+    truststore_file: "truststore.bcfks"
+    truststore_pass: readonlyrest
+```
+3. In case you are using ES >= 7.10 you need to modify `$JAVA_HOME/conf/security/java.policy` file and add this section at the end of it. It is required because ES restricted permissions that could be used by plugins.
+```
+grant {
+  permission org.bouncycastle.crypto.CryptoServicesPermission "exportSecretKey";
+  permission org.bouncycastle.crypto.CryptoServicesPermission "exportPrivateKey";
+  permission java.security.SecurityPermission "getProperty.jdk.tls.disabledAlgorithms";
+  permission java.security.SecurityPermission "getProperty.jdk.certpath.disabledAlgorithms";
+  permission java.security.SecurityPermission "getProperty.keystore.type.compat";
+  permission java.security.SecurityPermission "removeProvider.SunRsaSign";
+  permission java.security.SecurityPermission "removeProvider.SunJSSE";
+  permission java.io.FilePermission "${java.home}/lib/security/jssecacerts", "read";
+  permission java.io.FilePermission "${java.home}/lib/security/cacerts", "read";
+  permission java.security.SecurityPermission "getProperty.jdk.tls.server.defaultDHEParameters";
+  permission org.bouncycastle.crypto.CryptoServicesPermission "defaultRandomConfig";
+};
+```
+
+
+## How to convert JKS/PKCS12 keystore files into BCFKS
+
+1. Download [jar with bc-fips](https://repo1.maven.org/maven2/org/bouncycastle/bc-fips/1.0.2.3/bc-fips-1.0.2.3.jar) library and place it preferably in the same directory where you store keystore files to convert. 
+
+2. Open terminal and go to directory with keystore to convert
+
+3. Use keytool with following parameters to perform the conversion:
+  ```
+  keytool \
+  -importkeystore \
+  -srckeystore < filename of keystore that you want to convert >  \
+  -destkeystore < name of output keystore, typically should end with .bcfks > \
+  -srcstoretype < type of input keystore, should be JKS or PKCS12 > \
+  -deststoretype BCFKS \
+  -deststorepass < password for input keystore > \
+  -srcstorepass < password for output keystore > \
+  -providerpath ./bc-fips-1.0.2.1.jar \
+  -provider org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider
+  ```
+  If you placed dowloaded jar with bc-fips then you have to use appropriate path instead of `./bc-fips-1.0.2.1.jar` 
```

Comment:
Typo + Remember to use articles in English ;)
However, you could just say:

 "If you saved the bc-fips jar in a different path, remember to run it using the appropriate path instead of `./bc-fips-1.0.2.1.jar`."

Reply:
Articles are the thing that I really struggle with :D I'm becoming better, but I still sometimes forget about them

Reply:
haha yes it's actually a thing: at various degrees, I noticed this in all of you Polish contributors. It must be something related to Polish language? Do you have/use articles? As a native neo latin language speaker, I sometimes give them for granted, but I appreciate it could be different for other people with different background.

Reply:
No, Polish doesn't have any form of articles. This is definitely a reason why Poles are not too good at using them :D I found this page with articles usage map https://en.wikipedia.org/wiki/Article_(grammar)#Crosslinguistic_variation I didn't even know that, so many European languages are using them

Reply:
AFAIK all Slavic languages don't have articles